docs: add events examples to the documentation (#526)
FuelLabs/fuels-wallet: packages/docs/docs/api.mdx

  file="../../sdk/src/Fuel.test.ts"
  testCase="User getProvider on fuels-ts Wallet"
/>

## Events

The Fuel client emits events when certain actions occur. These events can be listened to by using the `on` method.

### Usage

The `fuel` object has an `events` property which is an enum of all the events that can be listened to. <br />
The `on` method takes two arguments, the event name and a callback function. The callback function is passed the data associated with the event.

### Connection

Listen for a new connection to a node. Callback function receives the `connection` object.

```js
fuel.on(fuel.events.CONNECTION, async (connection) => {});

> ```

Reviewer: You can also import this snippets from the source code

### Accounts

> We can listen for when the client connects his accounts. The callback is passed the account object.

Reviewer 2: same changes as other comment
```suggestion
Listen for when the client connects his accounts. Callback function receives the `accounts addresses`.
```

```js

> fuel.on(fuel.events.ACCOUNTS, async (account) => {});

Reviewer 2: ```suggestion
fuel.on(fuel.events.ACCOUNTS, async (accounts) => {});
```

```

### Current Account

> We can listen for when the client is authorized to use an account. If there are multiple accounts, the current account on the wallet is returned. The callback is passed the account object.

Reviewer 2: copy change
```suggestion
Listen for when the current account is changed in the Fuel Wallet extension. Callback function receives `account address`.

> **Note**
> Event will only be fired if the current account is connected to DAPP
```

```js
fuel.on(fuel.events.CURRENT_ACCOUNT, async (account) => {});
```

### Network

> We can listen for when the client connects to a network. The callback is passed the network object.

Reviewer 2: ```suggestion
Listen for when the client connects to a network. The callback receives the `network` object.
```

  file="../../sdk/src/Fuel.test.ts"
  testCase="User getProvider on fuels-ts Wallet"
/>

## Events

> The Fuel client emits events when certain actions occur. These events can be listened to by using the `on` method.

Reviewer: ```suggestion
Fuel emits events when certain actions occur. These events can be listened to by using the `on` method.

The events API follows the native Node.js `EventEmitter` enabling, `on`, 'once', and 'off'.
```

### Usage

The `fuel` object has an `events` property which is an enum of all the events that can be listened to. <br />

> The `on` method takes two arguments, the event name and a callback function. The callback function is passed the data associated with the event.

Reviewer: ```suggestion
The `on` method takes two arguments, the event name and a callback function. The callback function receives data associated with the event.
```

### Connection

> Listen for a new connection to a node. Callback function receives the `connection` object.

Reviewer: ```suggestion
This event is triggered when the connection status change between the Wallet and the application. The callback function receives the `connection` object.
```

<CodeImport
  file="../src/hooks/useIsConnected.tsx"
  lineStart="10"
  lineEnd="28"
/>

### Accounts

> Listen for when the client connects his accounts. Callback function receives the `account` object.

Reviewer: ```suggestion
This event is triggered when the list of connected accounts to the application change. The callback function receives the `account` string.
```

<CodeImport
  file="../examples/events/Accounts.tsx"
  lineStart="24"
  lineEnd="33"
/>

### Current Account

> Listen for when the current account is changed in the Fuel Wallet Extension. Callback function receives `account` address.

Reviewer: ```suggestion
This event is triggered when the current account is changed in the Fuel Wallet Extension. The callback function receives the `account` address.

This event is only triggered if the application has access to the current account address. Otherwise, it is not triggered.
```

<CodeImport
  file="../examples/events/CurrentAccount.tsx"
  lineStart="24"
  lineEnd="35"
/>

### Network

> Listen for when the client connects to a network. Callback function receives the `network` object.

Reviewer: ```suggestion
This event is triggered when the client connects to a different network. The callback function receives the `network` object.
```

  file="../../sdk/src/Fuel.test.ts"
  testCase="User getProvider on fuels-ts Wallet"
/>

## Events

Fuel emits events when certain actions occur. These events can be listened to by using the `on` method.

> The events API follows the native Node.js `EventEmitter` enabling, `on`, 'once', and 'off'.

Reviewer: ```suggestion
The events API follows the native Node.js `EventEmitter` enabling, `on`, `once`, and `off`.
```
